### PR TITLE
Untrusted runtime subordinates

### PR DESCRIPTION
**Pull request by joedborg**
To allow use of https://github.com/charmed-kubernetes/charm-kata

### CODE REVIEW

**Comment by joedborg** [comment]
[First class Kata containers](https://trello.com/c/U24zJal4/1313-first-class-kata-containers)

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Charm for Containerd
 
 This subordinate charm deploys the [Containerd](https://containerd.io/)
-engine within a running Juju charm application. Containerd is an open platform
+engine within a running Juju charm model. Containerd is an open platform
```
Comment (hyperbolic2346) [inline README.md]: charm model seems awkward. I would just say Juju model.

File: reactive/containerd.py [hunk 2]
```diff
@@ -275,6 +276,17 @@ def config_changed():
     context['custom_registries'] = \
         merge_custom_registries(context['custom_registries'])
 
+    untrusted = DB.get('untrusted')
+    if untrusted:
+        context['untrusted'] = True
+        context['untrusted_name'] = untrusted['name']
+        context['untrusted_path'] = untrusted['binary_path']
```
Comment (hyperbolic2346) [inline reactive/containerd.py]: If binary_path has the full path including the executable, is it wrong to call this untrusted_path? This is obviously a nitpick type thing, but is it a path if it has an executable in it? I'm honestly not sure, but my assumption was this was a path without an executable name in it. like `/usr/local/bin` and not `/usr/local/bin/my_exe`. Maybe I just need to breathe more...

File: reactive/containerd.py [hunk 3]
```diff
@@ -358,6 +370,37 @@ def publish_config():
...
+
+@when('endpoint.untrusted.departed')
+def untrusted_departed():
+    """
+    Handle untrusted container runtime.
+
+    :return: None
+    """
+    DB.unset('untrusted')
+    DB.flush()
```
Comment (hyperbolic2346) [inline reactive/containerd.py]: we don't flush when we set, why do we need it here?

Comment (joedborg) [reply reactive/containerd.py]: I saw it on some other code and asked in IRC.  I'm trying to remember why, but there is a reason that unset can be async and needs a flush.

File: reactive/containerd.py (hunk 2 above)
Comment (joedborg) [reply reactive/containerd.py]: I see where you're coming from, but I think a path can be a dir or file, no?

File: templates/config.toml [hunk 4]
```diff
@@ -35,16 +35,23 @@ oom_score = 0
     enable_tls_streaming = false
     max_container_log_line_size = 16384
     [plugins.cri.containerd]
-      snapshotter = "overlayfs"
```
Comment (hyperbolic2346) [inline templates/config.toml]: Are we sure we want to remove this line? Just verifying.

Comment (joedborg) [reply templates/config.toml]: Yeah, it's not in the recommended default config anymore (which is very transient).

**Pull request merged**